diff --git a/config.def.h b/config.def.h
index 9f7baac..761cce7 100644
--- a/config.def.h
+++ b/config.def.h
@@ -22,7 +22,7 @@ static const char *colorname[NUMCOLS] = {
 
 #if MESSAGE_PATCH || COLOR_MESSAGE_PATCH
 /* default message */
-static const char * message = "Suckless: Software that sucks less.";
+static const char * message = "Locked.";
 
 /* text color */
 static const char * text_color = "#ffffff";
@@ -86,7 +86,7 @@ ResourcePref resources[] = {
 
 #if ALPHA_PATCH
 /* lock screen opacity */
-static const float alpha = 0.9;
+static const float alpha = 0.5;
 #endif // ALPHA_PATCH
 
 /* treat a cleared input like a wrong password (color) */
@@ -137,7 +137,7 @@ static const int controlkeyclear = 0;
 
 #if DPMS_PATCH
 /* time in seconds before the monitor shuts down */
-static const int monitortime = 5;
+static const int monitortime = 30;
 #endif // DPMS_PATCH
 
 #if KEYPRESS_FEEDBACK_PATCH
@@ -161,4 +161,4 @@ static const char* pam_service = "login";
 #if QUICKCANCEL_PATCH
 /* time in seconds to cancel lock with mouse movement */
 static const int timetocancel = 4;
-#endif // QUICKCANCEL_PATCH
\ No newline at end of file
+#endif // QUICKCANCEL_PATCH
diff --git a/config.mk b/config.mk
index 6299928..ef7c751 100644
--- a/config.mk
+++ b/config.mk
@@ -17,8 +17,8 @@ X11LIB = /usr/X11R6/lib
 #NETBSD=-D_NETBSD_SOURCE
 
 # Uncomment for message patch / MESSAGE_PATCH / COLORMESSAGE_PATCH / DWM_LOGO_PATCH
-#XINERAMA=-lXinerama
-#XINERAMAFLAGS = -DXINERAMA
+XINERAMA=-lXinerama
+XINERAMAFLAGS = -DXINERAMA
 
 # Uncomment for pam auth patch / PAMAUTH_PATCH
 #PAM=-lpam
